persist and show containerService type and portConfigs.nodePort

moolitayer opened the pull request:
serviceSpec.type from k8s. Since rails uses the 'type'
columns for inheritance it is named 'service_type' internally.
Possible values in k8s: clusterIp, nodePort, loadBalancer.

display in container service views, both at item and explorer.

kubernetes: persist service port condig nodePort

display in container service item view in port config table.

reorder columns to match k8b v1 types.go order.

moolitayer commented:
fyi @abonas @simon3z 
@miq-bot add_label providers/containers enhancement ui

moolitayer commented:
item view:
![screenshot from 2015-08-30 14-20-20](https://cloud.githubusercontent.com/assets/3010449/9567104/456916d8-4f27-11e5-813c-c7d19df18f39.png)

explorer view:
![screenshot from 2015-08-30 14-23-38](https://cloud.githubusercontent.com/assets/3010449/9567106/4b9e5bc6-4f27-11e5-8e4b-818e4441e291.png)

moolitayer commented:
@miq-bot add_label providers/containers, enhancement, ui

moolitayer commented:
added a test

moolitayer commented:
> Checked commits moolitayer@eaa2f99...583e2e5 with rubocop 0.33.0 and haml-lint 0.13.0
> 3 files checked, 3 offenses detected
> 
> app/helpers/container_service_helper/textual_summary.rb
> 
>   Error - Line 8, Col 7 - Syntax - unexpected token tSTRING_CONTENT
>   Error - Line 32, Col 8 - Syntax - unexpected token tSTRING_CONTENT
>   Error - Line 79, Col 1 - Syntax - unexpected token $end

false positive?

abonas commented:
IMO the selected name ingress is confusing both in the backend and even more in the UI.
How can a user understand what is this field standing for?
Based on this https://github.com/kubernetes/kubernetes/blob/093614ec324c56b227de3f5d47dc12f88db3c2b9/docs/user-guide/services.md#type-loadbalancer
the 'ingress' term refers only to the load balancer type, and is an internal section containing other information.
So I recomment the name to be changed to something more clear that will explain what it is about. (or align with k8s terminology for it)

moolitayer commented:
Renamed column, verified, updated commit messages and pr description, uploaded new screenshots.

moolitayer commented:
added patches:
reorder, persist & show node port (see https://github.com/ManageIQ/manageiq/pull/4112/commits)

abonas commented on db/migrate/20151435234635_add_node_port.rb:
```diff
@@ -0,0 +1,5 @@
+class AddNodePort < ActiveRecord::Migration
+  def change
+    add_column :container_service_port_configs, :node_port, :string
```
shouldn't port be an integer?

moolitayer replied:
fixed, thanks

blomquisg commented:
@matthewd do you know what these rubocop failures mean?  I couldn't actually find anything wrong with those first two error when just running them in irb.

> **app/helpers/container_service_helper/textual_summary.rb**
> - [ ] :red_circle: **Error** - [Line 8](https://github.com/moolitayer/manageiq/blob/31ab722ad859cd7b137913f51f393db9af872f8c/app/helpers/container_service_helper/textual_summary.rb#L8), Col 7 - Syntax - unexpected token tSTRING_CONTENT
> - [ ] :red_circle: **Error** - [Line 33](https://github.com/moolitayer/manageiq/blob/31ab722ad859cd7b137913f51f393db9af872f8c/app/helpers/container_service_helper/textual_summary.rb#L33), Col 8 - Syntax - unexpected token tSTRING_CONTENT
> - [ ] :red_circle: **Error** - [Line 80](https://github.com/moolitayer/manageiq/blob/31ab722ad859cd7b137913f51f393db9af872f8c/app/helpers/container_service_helper/textual_summary.rb#L80), Col 1 - Syntax - unexpected token $end

chessbyte commented on app/helpers/container_service_helper/textual_summary.rb:
```diff
@@ -9,19 +9,21 @@ def textual_group_properties
 
   def textual_group_port_configs
-    labels = [_("Name"), _("Port"), _("Target Port"), _("Protocol")]
+    labels = [_("Name"), _("Port"), _("Target Port"), _("Protocol"), ("Node Port")]
```
Why is `("Node Port")` without a leading underscore like the other items in this array?

chessbyte commented:
@moolitayer besides my comment about I18N, this PR looks good to me.
@abonas @simon3z is this PR ready to merge?

simon3z commented:
@moolitayer please squash all the patches in 1, this change is small enough.

chessbyte commented:
@simon3z what is the value of squashing commits here?

simon3z commented on app/helpers/container_service_helper/textual_summary.rb:
```diff
@@ -9,19 +9,21 @@ def textual_group_properties
         config.target_port,
-        config.protocol
+        config.protocol,
+        config.node_port
```
@moolitayer would it make sense to keep the same order here as well?

simon3z commented:
> @simon3z what is the value of squashing commits here?

@chessbyte the feature is "collect and display nodePort" and having 6 patches to:

- collect service type
- display service type
- display service type in the explorer
- reorder 2 entries in an hash
- collect node port
- display node port

seems overkilling imo (also if you consider that each patch has ~5 lines changed). Maybe splitting service type and node port is a good idea (2 patches).

chessbyte commented:
@simon3z deferring squash decision to you.  2 commits seems like a reasonable idea.

jrafanie commented:
The migrations look to be named wrong, see 14th month. Please check with @fryguy and @matthewd if you need help. Fryguy recently fixed several migrations that were named incorrectly.

moolitayer commented:
fixed I18N issue + migration script dates. 

@simon3z:
* added display in app/helpers/container_service_helper/textual_summary.rb to reorder patch
* why do you want to squash?  I did one commit per change as we were specifically requested before. 
why does it matter if a change is only x lines long if it's ? Let me know what you decide on this one.

chessbyte commented:
@simon3z @abonas Please review

simon3z commented:
@chessbyte I'll review this ASAP (although this is not critical for the milestone of the 14th, compared to other things @moolitayer has on the table)

simon3z commented on app/helpers/container_service_helper/textual_summary.rb:
```diff
@@ -9,19 +9,21 @@ def textual_group_properties
     h[:values] = @record.container_service_port_configs.collect do |config|
       [
         config.name || _("<Unnamed>"),
+        config.protocol,
```
@moolitayer why this change? Isn't placing the `protocol` in the wrong column?

moolitayer replied:
This was supposed to be part of [1] but I forgot to reorder the labels. Fixed.

[1] "reorder columns to match k8b v1 types.go order."
Name, Protocol, Port, Target Port

simon3z commented:
@blomquisg @chessbyte LGTM :+1: ready for merge
